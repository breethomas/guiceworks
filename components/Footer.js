// @flow
import React from 'react'
import css, { media2 } from '../styles/css'
import { maxWidthContent } from '../styles/jso'
import Copy from './Copy'
import Title from './Title'
import View from './View'

const currentYear = new Date().getFullYear()

const footerStyle = css(
  {
    display: 'flex',
    flexFlow: 'column wrap',
    marginTop: 'auto',
    backgroundColor: '#fff',
    minHeight: 340,
  },
  media2({
    minHeight: 540,
  }),
)

const contactStyle = css({
  display: 'flex',
  flexFlow: 'column wrap',
  justifyContent: 'center',
  flex: 1,
  width: '100%',
})

const contentInfoStyle = css({
  display: 'flex',
  flexFlow: 'column wrap',
  justifyContent: 'center',
  width: '100%',
  minHeight: 85,
  marginTop: 'auto',
  color: '#fff',
  backgroundColor: '#010101',
})

const innerStyle = css(
  maxWidthContent,
)

const offColorStyle = css({
  color: '#808080',
})

const mailStyle = css({
  color: '#010101',
  borderBottom: '1px solid #4d4d4d',
  textDecoration: 'none',
})

const MailLink = () => (
  <strong>
    <a className={mailStyle} href="mailto:info@guiceworks.com">
      info@guiceworks.com
    </a>
  </strong>
)

export default () => (
  <footer className={footerStyle}>
    <View className={contactStyle}>
      <View className={innerStyle}>
        <Title tag="h2">Contact.</Title>
        <View className={offColorStyle}>
          <Copy isThin >
            For more information or work samples email us at <MailLink />
          </Copy>
        </View>
      </View>
    </View>
    <View className={contentInfoStyle} role="contentinfo">
      <View className={innerStyle}>
        <Copy isSmall isThin >
          <span>&copy;</span>
          <span>{` Copyright ${currentYear} Guiceworks INC All rights reserved.`}</span>
        </Copy>
      </View>
    </View>
  </footer>
)

