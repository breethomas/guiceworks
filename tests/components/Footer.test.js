import React from 'react'
import renderer from 'react-test-renderer'
import Footer from '../../components/Footer'

describe('Footer', () => {
  let component

  beforeEach(() => {
    component = <Footer />
  })

  it('shows the correct snapshot for the Footer component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
