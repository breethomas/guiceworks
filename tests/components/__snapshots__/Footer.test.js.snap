// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer shows the correct snapshot for the Footer component tree 1`] = `
<footer
  className={
    Object {
      "data-css-p8642l": "",
    }
  }
>
  <div
    className={
      Object {
        "data-css-q392fl": "",
      }
    }
  >
    <div
      className={
        Object {
          "data-css-24spax": "",
        }
      }
    >
      <h2
        className="css-lcomo css-5eu6oy"
      >
        Contact.
      </h2>
      <div
        className={
          Object {
            "data-css-1cicj1j": "",
          }
        }
      >
        <p
          className="css-isubv6 css-1eyhu5m"
        >
          For more information or work samples email us at 
          <strong>
            <a
              className={
                Object {
                  "data-css-1wegod6": "",
                }
              }
              href="mailto:info@guiceworks.com"
            >
              info@guiceworks.com
            </a>
          </strong>
        </p>
      </div>
    </div>
  </div>
  <div
    className={
      Object {
        "data-css-1hxbxk7": "",
      }
    }
    role="contentinfo"
  >
    <div
      className={
        Object {
          "data-css-24spax": "",
        }
      }
    >
      <p
        className="css-isubv6 css-1jmoywn"
      >
        <span>
          Â©
        </span>
        <span>
           Copyright 2018 Guiceworks INC All rights reserved.
        </span>
      </p>
    </div>
  </div>
</footer>
`;
